modules/ROOT/pages/apikit-odatav4-expand-tutorial.adoc

```diff
@@ -0,0 +1,162 @@
+differs: once the information for the main entity set is retrieved, the module
+will perform additional _virtual_ requests gathering the information for any
+`expand`-ed fields. If errors arise during this expansion operation, the final
+behaviour is configurable (throw the error to the consumer or ignore it
+resolving a `null` value).
+
+== Requirements
+
+- APIKit for OData Module (version 1.3.0)
+- An `<entity-collection-source/>` implementation for each expandable entity.
+- Support for the `in` filter on those `<entity-collection-source/>` flows.
+
+== Implementation guidelines
+
+In order to use our generic expansion implementation each source has to be
+annotated with the relevant types that might be expanded from it. Keep in mind
+that enabling `$expand` in a completely unrestricted way will allow API clients
+to write arbitrarily complex queries, this can cause performance problems.
+
+If you wish to implement `$expand` by yourself for some relationships leave
+those out of the list on the "expandable navigation properties" option. Manual
+implementations inside the flow will be needed for navigation properties that
+don't use _generic expansion_.
+
+=== Implementation checklist
+
+. Check that your CSDL and Mule application meets the expected requirements:
+** Every `NavigationProperty` should have one `ReferentialConstraint`.
+** Every `EntitySet` should have one `NavigationPropertyBinding` for each
+   `NavigationProperty` of its entity type.
+** Every `<entity-collection-source/>` should support the `in` operator for
+   the `$filter` system query option.
+. For each source where you want to enable _generic expand_ support:
+.. Open the *Capabilities* tab
+.. Edit inline *Expandable navigation properties*
+.. Add the name of each navigation property you want to be expandable using
+   _generic expand_.
+.. Depending on your use case check *Ignore errors on expand* if you don't want
+   errors during generic expansion to generate errors on the main flow.
+. Enjoy how your existent flows enable `$expand` support for you!
+
+== How _generic expand_ works
+
+Generic expand works by dispatching additional queries to the application once
+the original flow is completed. Because this is done based on the application
+`EDM` extra annotations might be needed in order to add enough context to the
+module:
+
+. The `EntitySet` where the expansion takes place should have a
+  `NavitationPropertyBinding` for every expandable `NavigationProperty` of its
+  `Entity`. +
+  This used to select which `<entity-collection-source/>` will be executed when
+  expanding that property.
+. Each expandable `NavigationProperty` should have a `ReferentialConstraint`
+  attached. +
+  This is used to create the inner query where the value from the `Property`
+  will be matched against the value of the `ReferencedProperty`.
+
+=== Flow execution
+
+. The main flow is executed and completed. A collection of _entities_ is its
+  result.
```

Comment:
The main flow is executed and completed, and it results in a collection of _entities_.